libgadget/blackhole.c

```diff
@@ -224,6 +224,20 @@ blackhole(ForceTree * tree, double * TimeNextSeedingCheck)
     N_sph_swallowed = N_BH_swallowed = 0;
 
     /* Let's determine which particles may be swalled and calculate total feedback weights */
+    SPH_SwallowID = mymalloc("SPH_SwallowID", SlotsManager->info[0].size * sizeof(MyIDType));
```

Comment:
Is this a bug fix? Seems to be related to Yueying's Ptype=5 Mass not tracking BlackholeMass issue.

Reply:
I don't think it is a bug fix - it is just allocating this memory within blackhole.c instead of in sph_particle_data

Reply:
It might be that there is a bug there, but this does not change it as far as I can see.